pennylane/_queuing.py

```diff
@@ -61,22 +61,33 @@ def __exit__(self, exception_type, exception_value, traceback):
         QueuingContext._active_contexts.remove(self)
 
     @abc.abstractmethod
-    def _append(self, obj):
+    def _append(self, obj, **kwargs):
         """Append an object to this QueuingContext instance.
 
         Args:
             obj: The object to be appended
         """
 
     @classmethod
-    def append(cls, obj):
+    def active_context(cls):
+        """Returns the currently active queuing context."""
+        # TODO: this method should append only to `cls.active_context`, *not*
+        # all active contexts. However this will require a refactor in
+        # the template decorator and the operation recorder.
+        if cls._active_contexts:
+            return cls._active_contexts[-1]
+
+        return None
+
+    @classmethod
+    def append(cls, obj, **kwargs):
         """Append an object to the queue(s).
 
         Args:
             obj: the object to be appended
```

Comment:
Worth mentioning anything about kwargs here?

Reply:
It's hard to, as the QueuingContext doesn't know/care about these kwargs, it's left for a subclass to define.